File: src/main/java/upbrella/be/user/dto/response/UserInfoResponse.java
package upbrella.be.user.dto.response;

import lombok.Builder;
import lombok.Getter;
import upbrella.be.user.entity.User;

@Getter
@Builder
public class UserInfoResponse {

    private long id;
    private String name;
    private String phoneNumber;
    private String bank;
    private String accountNumber;
    private String email;

    public static UserInfoResponse fromUser(User user) {

        return UserInfoResponse.builder()
                .id(user.getId())
                .name(user.getName())
                .phoneNumber(user.getPhoneNumber())
                .bank(user.getBank())
                .accountNumber(user.getAccountNumber())
                .email(user.getEmail())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/user/dto/response/UserInfoResponse.kt
package upbrella.be.user.dto.response

import upbrella.be.user.entity.User

data class UserInfoResponse(
    val id: Long,
    val name: String,
    val phoneNumber: String?,
    val bank: String?,
    val accountNumber: String?,
    val email: String,
    val adminStatus: Boolean
) {
    companion object {
        @JvmStatic
        fun fromUser(user: User): UserInfoResponse {
            return UserInfoResponse(
                id = user.id!!,
                name = user.name,
                phoneNumber = user.phoneNumber,
                bank = user.bank,
                accountNumber = user.accountNumber,
                email = user.email,
                adminStatus = user.adminStatus
            )
        }
    }
}
